libsolidity/formal/SMTEncoder.cpp

```diff
@@ -515,14 +583,16 @@ void SMTEncoder::endVisit(FunctionCall const& _funCall)
 void SMTEncoder::initFunction(FunctionDefinition const& _function)
 {
 	solAssert(m_callStack.empty(), "");
+	solAssert(m_currentContract, "");
 	m_context.reset();
 	m_context.pushSolver();
 	m_pathConditions.clear();
 	pushCallStack({&_function, nullptr});
 	m_uninterpretedTerms.clear();
-	resetStateVariables();
-	initializeLocalVariables(_function);
+	createStateVariables(*m_currentContract);
+	createLocalVariables(_function);
```

Comment:
Now variables are only created, that is, the symbolic variables are created, but no constraints are added at this point.